app/components/common/textBlurb/textBlurbDirective.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * textBlurbDirective.js
+ *
+ * A text box with a heading, and a read more link that expands the box to show the full contents
+ **/
+(function() {
+  'use strict';
+  
+  textBlurbDirective.$inject = ['$window'];
+  function textBlurbDirective($window) {
+    var directive = {
+      restrict: 'E',
+      bindToController: {
+        title: '@',
+        body: '@',
+      },
+      link: link,
+      controller: function() {},
+      controllerAs: 'textBlurb',
+      template: require('./textBlurb.html')
+    };
+
+    return directive;
+
+    //////////////////////////////
+    function link(scope, element, attrs, textBlurb) {
+      textBlurb.isCollapsed = true;
+      
+      scope.$watch(function() {return textBlurb.body; }, refreshButtons);
+      angular.element($window).bind('resize', refreshButtons);
+
+      function refreshButtons() {
+        textBlurb.showButtons = shouldShowButtons();
+        scope.$digest(); //since you're outside of angular, you need this here to update the view 
```

Comment:
This is because of the event handling. Anytime you're calling a handler from an event (typing, clicking, resizing) or doing something async, it's not going to be called within the angular context.